### ISSUE TITLE
Add keymap, embark instructions, etc for buffer (at-point) actions

### ISSUE THREAD

**Issue by bdarcus** [bdarcus/bibtex-actions#142]
This can be a killer feature in general, and when coupled with `org-cite`, and would feature to include for the 1.0 release.

In effect it would provide a super flexible kind of embarkified micro-org-ref (emphasis on "micro" though); or at least a framework for that to evolve.

What do we need?

1. a keymap specific to this case; maybe `bibtex-actions-buffer-map`? It would obviously include all the "open" commands; not sure what else.
2. README content for this in configuration and usage, in part to promote it (once org-cite is merged, I'll want to mention support for it prominently at the top, and include a screenshot), in part to make it easy for people to get working

Do we need any actual code here, or just code examples in the README? 

For example, @publicimageltd mentioned [here](https://github.com/bdarcus/bibtex-actions/pull/137#issuecomment-857649435): 

> ... would be cool, of course, to have some kind of automatic plug in, e.g. with eval-after-load. 

Not sure the idea here; would it apply to embark-minibuffer too? As in, rather than document the config in the README, we'd set it up for people if they have embark installed?

Also, now understanding all this, and recognizing the limitations of current CRM UIs, **I think we need an option, maybe default, to run at-point functions directly, without further user interaction**.

I'll be traveling for a couple of weeks starting tomorrow (though will still have my computer and access; just maybe not as much), so thought I'd just start this here, while ideas are fresh. If someone wants to start a PR, please feel free.

See also #141 and #140.

cc @ilupin

**Comment by publicimageltd** [comment]
Good idea! I am not sure I understand all questions or possibilities listed above. So here's my vision:
 - provide a keymap for all "open" actions, which can then be used for embark
 - provide some code which would hook these actions in embark, combined with the target finder, once embark is (also) loaded (the `with-eval-after-load` idea)
 - Provide some functions or functionality to call all open functions also directly "on point". This could be done either via "initial input" in the current functions, or by providing specific functions which then use the initial input.

Is this what we are talking about? I am currently also more offline, RL is using up my time; but if there is a clear roadmap, I'd like to contribute some of this logic. Or at least I'd continue commenting stuff without even updating my local git repo :smile:

**Comment by bdarcus** [comment]
Sorry; I sometimes get a bit wordy as I'm trying to sort things out in my head, and am a bit distracted. But ...

> Is this what we are talking about? 

But yes @publicimageltd, that sounds perfect to me!

One little amendment:

> This could be done either via "initial input" in the current functions, or by providing specific functions which then use the initial input.

One of the pieces I mention above is that initial input does not work well with multiple candidate selection. I don't see why we don't just entirely bypass completing-read here. If I want to open a PDF for a citation reference, should I just do it directly?

> I am currently also more offline, RL is using up my time; but if there is a clear roadmap, I'd like to contribute some of this logic.

WDYT @ilupin?

**Comment by bdarcus** [comment]
> I don't see why we don't just entirely bypass completing-read here. If I want to open a PDF for a citation reference, shouldn't I just do it directly?

In the end, the details will matter and require UX testing, but another idea: run the function directly per above, but fallback to completing-read if no result?

**Comment by ilupin** [comment]
I'm using embark and I strongly agree that completing-read is not necessary for at-point actions. I will submit a PR to add more embark support for bibtex-actions in the next few days.

**Comment by publicimageltd** [comment]
> > I don't see why we don't just entirely bypass completing-read here. If I want to open a PDF for a citation reference, shouldn't I just do it directly?
> 
> In the end, the details will matter and require UX testing, but another idea: run the function directly per above, but fallback to completing-read (the minibuffer) if no result?

That seems like a good compromise; sometimes keys got changed, or there's a typo which can be resolved easily. But as to per the UX, it might feel odd to want to open something, and then get a list of choices. If one adds initial input, the list will be most likely empty, so it will be most likely a full list. But that is what I take to be odd: oops, I didn't want that list, where's my PDF I just wanted to open? So that would require some kind of extended prompt, like "Cannot find reference at point; select PDF to open:". Which, in turn, would mean to have different prompts for the open functions according to their calling context.

**Comment by bdarcus** [comment]
@publicimageltd 

> Which, in turn, would mean to have different prompts for the open functions according to their calling context.

... which probably means my strawmsn was a bad idea.

My preference is to treat org-cite as the primary for this functionality, because it provides the maximal framework for it.

For example, an "activate" processor could, in theory, somehow add information about presence of associated PDF, link, note to a citation-reference.

**Comment by bdarcus** [comment]
Just to succinctly explain why not use completing-read here:

We use bibtex-actions-read to find the key(s), and act in them.

Here we already have the key(s).

**Comment by bdarcus** [comment]
I believe this was addressed in #143.
